src/core/control/tools/ImageHandler.h

```diff
@@ -26,16 +26,42 @@ class ImageHandler {
     virtual ~ImageHandler();
 
 public:
-    bool insertImage(double x, double y);
-    // does the complete insertion based on create and add
-    bool insertImage(GFile* file, double x, double y);
-    // creates the image and does automatic scaling of the image
-    std::tuple<Image*, int, int> createImage(GFile* file, double x, double y);
-    bool addImageToDocument(Image* img, bool addUndoAction);
-    // scale down (only if necessary) the image so that the it fits on the page
-    // applies (potentially adjusted) width/height to the image
+    /**
+     * inserts an image scaled to the given size
+     */
+    auto insertImageWithSize(xoj::util::Rectangle<double> space) -> bool;
+
+    /**
+     * lets the user choose an image file and then creates the image
+     */
+    auto chooseAndCreateImage(double x, double y) -> std::tuple<Image*, int, int>;
+
+    /**
+     * creates the image from the given file
+     */
+    auto createImageFromFile(GFile* file, double x, double y) -> std::tuple<Image*, int, int>;
+
+    auto addImageToDocument(Image* img, bool addUndoAction) -> bool;
+
+    /**
+     * scale down (only if necessary) the image so that it then fits on the page
+     * applies (potentially adjusted) width/height to the image
+     */
     void automaticScaling(Image* img, double x, double y, int width, int height);
 
+    /**
+     * scales image down to fit given space
+     *
+     * this will not stretch the image
+     * if the available space is 0, no scaling occurs
+     */
+    static void scaleImageDown(Image* img, xoj::util::Rectangle<double> space);
```

Comment:
```suggestion
    static void scaleImageDown(Image* img, const xoj::util::Rectangle<double>& space);
```

Reply:
Also, these static functions could be moved into a local function defined in the .cpp file.